torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -1937,22 +1937,25 @@ void TranslateApplicableWelford::translateSingleWelford(WelfordOp* welford) {
   auto out_N = welford->outN()->as<TensorView>();
 
   fusion->removeExpr(welford);
+  // Not safe to use welford anymore
+  welford = nullptr;
 
   // Create normalization based welford graph
   //  largely taken from batchnorm cpp benchmark
-  auto& in_root = in_val->getRootDomain();
-  auto& out_root = out_avg->getRootDomain();
+  const auto& in_root =
+      TensorDomain::noReductions(in_val->getMaybeRFactorDomain());
+  const auto& out_root = out_avg->getRootDomain();
```

Comment:
Can we put the assert here as well: https://github.com/pytorch/pytorch/pull/77010/files
Since it's not obvious the in_root.size() <= out_root.size() and we could go over range -> segfault.

Reply:
I changed `[i]` to `.at(i)`, so if that happens, we should see an explicit exception. It was a silent out-of-bounds error before. 

That said, I don't want to see the same error again, so I'll add the assertion too.